src/js/player.js

```diff
@@ -1873,12 +1873,37 @@ class Player extends Component {
    * Attempt to begin playback at the first opportunity.
    *
    * @return {Promise|undefined}
-   *         Returns a `Promise` only if the browser returns one and the player
-   *         is ready to begin playback. For some browsers and all non-ready
-   *         situations, this will return `undefined`.
+   *         Returns a promise if the browser supports Promises (or one
+   *         was passed in as an option). This promise will be resolved on
+   *         the return value of play. If this is undefined it will fulfill the
+   *         promise chain otherwise the promise chain will be fulfilled when
+   *         the promise from play is fulfilled.
    */
   play() {
+    const promise = this.options_.Promise || window.Promise;
 
+    if (promise) {
+      return new Promise((resolve) => {
+        this.play_(resolve);
+      });
+    }
+
+    return this.play_();
```

Comment:
ie11 does not support promises, so that should be the only platform this happens on.